staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -51,18 +50,31 @@ func NewOptions() *Options {
 }
 
 // Validate check LogFormat in registry or not
-func (o *Options) Validate() []error {
+// and gets the list of all unsupported flags
+// with all the unsupported flags that are set
+func (o *Options) Validate(fs *pflag.FlagSet) []string {
+	// First, check if the log format is available or not
 	if _, err := o.Get(); err != nil {
-		return []error{fmt.Errorf("unsupported log format: %s", o.LogFormat)}
+		return []string{fmt.Sprintf("unsupported log format: %s", o.LogFormat), ""}
+	}
+
+	return []string{fmt.Sprintf("unsupported logging flags set for logging format: %s, unsupported flags for non-default formats: %s\n", strings.Join(allUnsupportedLoggingFlags(), ", --")),
```

Comment:
Validate should fail only `if o.Options != defaultLogFormat` and `len(setUnsupportedLoggingFlags()) != 0`